File: practice21/4.cpp
//https://www.nowcoder.com/practice/a2d5b1875bb0408384278f40d1f236c9?tpId=90&tqId=30808&tPage=2&rp=2&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
#include<string>
using namespace std;

int main()
{
	int i, m, temp, length;
	string str;
	while (cin >> str)
	{
		length = 0, temp = 0;
		m = str.size();
		for (i = 0; i < m; i++)
		{
			if (str[i] == '(')
			{
				temp++;
				if (temp > length)
				{
					length = temp;
				}
			}
			else
			{
				temp--;
			}
		}
		cout << length << endl;
	}
	system("pause");
	return 0;
}

File: practice28/4.java


//https://www.nowcoder.com/practice/b6b63d3c0ff140a481b4f9acda922503?tpId=98&tqId=33027&tPage=11&rp=11&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.io.*;
import java.math.BigInteger;
import java.util.Scanner;

public class Main {
    public static void Calculation(long num){
        int count,temp,m;
        count=0;
        for(int i=1;i<=num;i++){
            temp=i;
            while(temp>0){
                m=temp/5;
                count=count+m;
                temp=temp/5;
            }
        }
        System.out.println(count);
    }
    public static void main(String args[]) throws IOException {
        Scanner scanner=new Scanner(System.in);
        long n=scanner.nextLong();
        Calculation(n);
    }
}